Documentation/app-operations.md

```diff
@@ -0,0 +1,197 @@
+```
+
+# Entrypoints
+
+In order to facilitate the app-level operations API, 4 new entrypoints are introduced.
+These entrypoints are stage1 binaries that are capable of communicating with the stage1 supervisor.
+
+For example, the `app start` entrypoint in default stage1 flavor sends a `start` signal
+(similar to `systemctl start`) for the service files of the application to the systemd which is the stage1 supervisor.
+
+## rkt app inject
+
+`coreos.com/rkt/stage1/app/inject`
+
+1. prepare the application intended for injection, its rootfs and service files or else.
+2. `coreos.com/rkt/stage1/inject` entrypoint is resolved via annotations found within `/var/lib/rkt/pods/run/$uuid/stage1/manifest`.
+3. the entrypoint will be executed and passed a reference to the rootfs of the application that was prepared.
+
+The responsibility of this entrypoint is to receive a prepared app, copy its contents into stage2, 
+so that it is ready to be started by the stage1 supervisor (e.g. systemd for the default flavor).
+
+### Stage1 Default Flavor Example Workflow
+
+1. Prepare the application rootfs in stage0 for the application image/images.
+2. Configure the application service files given an application image/images.
+3. Copy the application rootfs to `/var/lib/rkt/pods/run/$uuid/stage1/rootfs/opt/stage2/`.
+4. Copy the service files of the application to the stage1 units directory `/usr/lib/systemd/system`.
+5. Start the injected application using the `app start` entrypoint.
+
+## rkt app eject
+
+`coreos.com/rkt/stage1/app/eject`
+
+1. receive a reference to an application that resides inside the pod (running or stopped)
+2. stop the application if it's running.
+3. remove the contents of the application (rootfs) from the pod (keep the logs?) and delete references to it (e.g. service files).
+   - Delegate these steps by creating `app/install` and `app/eject` entrypoints?
+
+The responsibility of this entrypoint is to receive an app inside a pod and to remove it from the pod.
+so that it is ready to be started by the stage1 supervisor (e.g. systemd for the default flavor).
+After this operation, starting the application again is not possible and requires injecting the same
+application using the `app/inject` entrypoint.
+
+### Stage1 Default Flavor Example Workflow
+
+1. Stop the application using the `app stop` entrypoint.
+2. Delete the application rootfs in `/var/lib/rkt/pods/run/$uuid/stage1/rootfs/opt/stage2/`.
+3. Remove the application service files.
+4. Perform garbage collection during the execution of a pod.
+5. Reload the systemd daemon for changes to take place without disturbing the other tennant applications.
+
+## rkt app start
+
+`coreos.com/rkt/stage1/app/start`
+
+1. given a reference to an application that's in a startable state (see below), instruct the supervisor to start the application.
+
+The responsibility of this entrypoint is to start an application that is in a `non-running`, but also `non-ejected` state, 
```

Comment:
It might be useful to sketch out the possible states that an app can be in, defined either negatively or positively